### PR TITLE
Added wrapper for rsem-generate-data-matrix.

### PR DESCRIPTION
**Pull request by brcopeland**
### Description
This is a wrapper to summarize single-sample gene/isoform expression via RNA-seq using `rsem-calculate-expression` into a single table for downstream analysis.

### QC

For all wrappers added by this PR, I made sure that

* [x] there is a test case which covers any introduced changes,
* [x] `input:` and `output:` file paths in the resulting rule can be changed arbitrarily,
* [x] all `environment.yaml` specifications follow [the respective best practices](https://stackoverflow.com/a/64594513/2352071),
* [x] wherever possible, command line arguments are inferred and set automatically (e.g. based on file extensions in `input:` or `output:`),
* [x] all fields of the example rules in the `Snakefile`s and their entries are explained via comments (`input:`/`output:`/`params:` etc.),
* [x] `stderr` and/or `stdout` are logged correctly (`log:`), depending on the wrapped tool,
* [x] the `meta.yaml` contains a link to the documentation of the respective tool or command,
* [x] `Snakefile`s pass the linting (`snakemake --lint`),
* [x] `Snakefile`s are formatted with [snakefmt](https://github.com/snakemake/snakefmt),
* [x] Python wrapper scripts are formatted with [black](https://black.readthedocs.io).

### CODE REVIEW

**Comment by brcopeland** [comment]
Oh I guess I ran `black` on `test.py`, which introduced additional unexpected changes; but maybe it should be this way?  If not just let me know and I'll revert those.

File: bio/rsem/generate-data-matrix/test/Snakefile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+rule rsem_generate_data_matrix:
+    input:
+        # one or more expression files created by rsem-calculate-expression
+        "a.genes.results",
```
Comment (jafors) [inline bio/rsem/generate-data-matrix/test/Snakefile:4]: Could you add a second input file to better incorporate the typical use of this tool. Sorry for being a tad nitpicky here ;)

**Review by jafors — CHANGES_REQUESTED** [review]
Thanks again! Formatting the test.py does not seem problematic to me, should maybe be standard procedure with new tests.

Just a small request below.

File: bio/rsem/generate-data-matrix/test/Snakefile (hunk 1 above)
Comment (brcopeland) [reply bio/rsem/generate-data-matrix/test/Snakefile]: Sounds reasonable to me.  I have done that.

**Review by jafors — APPROVED** [review]
Great, thanks a lot!

**Pull request merged**